### ISSUE TITLE
Unmarshaling into struct pointer does not work, part 2

### ISSUE THREAD

**Issue by mikroskeem** [philandstuff/dhall-golang#24]
Borrowing code snippet from #9,

```golang
var config struct {
    Foo bool `json:"foo"`
    Bar *struct {
        Baz string `json:"baz"`
    } `json:"bar"`
}
```
and configuration
```dhall
{ foo = True }
```

should actually work as `Bar` field in first anonymous struct is a pointer type (which means that `Bar` should be `nil`), but instead I'll get `Don't know how to decode <nil> into struct`.

As mentioned in #9, then that works with 3rd party TOML and Golang's own JSON solution.

**Comment by mikroskeem** [comment]
But then again... Dhall has Optionals 🤔

**Comment by philandstuff** [comment]
Yeah I'm not sure that we should support unmarshalling partial records into Go structs like this.  The way I think about it is: you're trying to unmarshal a value whose type doesn't even match the go struct, so it should fail, though maybe the error message could be clearer.

I'd rather go with the Optional route here, which as far as I can see would work fine.

**Comment by mikroskeem** [comment]
Yeah Optional is the way, fits more with Dhall overall.

However I agree that the error message should be clearer
